### PR TITLE
Run all tests

### PR DESCRIPTION
**Pull request by jhelfferich**
With these modifications, the pipeline will run all Odoo tests on the branch. I have limited the tested modules to the ones installed in the JobRad production system.

### CODE REVIEW

File: tests/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+-i base,\
+account \
+--test-enable --test-tags=-post_install \
+--stop-after-init
+echo "---*** Running Odoo tests post_install ***---"
+docker compose -f tests/docker-compose.yml exec odoo ./odoo-bin -d test --addons-path=addons,odoo/addons \
+--test-enable --test-tags=/base\
+/account \
+--stop-after-init
+docker compose -f tests/docker-compose.yml down -v
```
Comment (borna-ng) [inline tests/run_tests.sh]: A little bit more robust and slightly more readable:

```suggestion
function cleanup() {
    docker compose -f tests/docker-compose.yml down -v
}

trap cleanup EXIT

docker compose -f tests/docker-compose.yml up -d

echo "---*** Running Odoo tests at_install ***---"
docker compose -f tests/docker-compose.yml exec odoo ./odoo-bin -d test --addons-path=addons,odoo/addons \
    -i base,account \
    --test-enable --test-tags=-post_install \
    --stop-after-init

echo "---*** Running Odoo tests post_install ***---"
docker compose -f tests/docker-compose.yml exec odoo ./odoo-bin -d test --addons-path=addons,odoo/addons \
    --test-enable --test-tags=/base,/account \
    --stop-after-init

docker compose -f tests/docker-compose.yml down -v
```

File: tests/run_tests.sh
Comment (borna-ng) [inline tests/run_tests.sh:1]: Shouldn't we have a list of modules that we actively use in production?

**Review by borna-ng — CHANGES_REQUESTED** [review]
Second `docker compose` command has an invalid tag specified `/base/account`. I assume it should be `/base,/account`.

https://github.com/jobrad-gmbh/odoo/actions/runs/11434716121/job/31809204451#step:4:1970

File: tests/run_tests.sh (hunk 1 above)
Comment (jhelfferich) [reply tests/run_tests.sh]: :thinking: This isn't the latest version of this file. I must have accidentally reverted that...

Nonetheless, great suggestions. I will incorporate them.

File: tests/run_tests.sh
Comment (jhelfferich) [reply tests/run_tests.sh:1]: Yes, and I used that. But I accidentally reverted back to my initial test, where I used just 2 modules to refine the script. Now, it is ready for re-testing.

File: pyproject.toml [hunk 2]
```diff
@@ -23,6 +23,7 @@ markupsafe = "1.1.0"
 num2words = "0.5.6"
 ofxparse = {version = "0.19", markers = "python_version <= \"3.9\""}
 passlib = "1.7.1"
+phonenumbers = "8.13.47"
```
Comment (borna-ng) [inline pyproject.toml]: If we need `phonenumbers` package only for testing I would suggest moving it into a separate group (e.g. `[tool.poetry.group.testing.dependencies]`) and making that group optional. We have a similar setup in [odoo-addons-jobrad](https://github.com/jobrad-gmbh/odoo-addons-jobrad/blob/354eabf55cb3f0032025a8b1aec19b303319b165/pyproject.toml#L107).

File: tests/run_tests.sh [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+trap cleanup EXIT
+
+docker compose -f tests/docker-compose.yml up -d
+
+echo "---*** Running Odoo tests at_install ***---"
+docker compose -f tests/docker-compose.yml exec odoo ./odoo-bin -d test --addons-path=addons,odoo/addons \
+  -i \
+base,\
+account,\
```
Comment (borna-ng) [inline tests/run_tests.sh]: It would probably make sense to load the list of modules from a file.

File: poetry.lock
Comment (borna-ng) [inline poetry.lock:1]: Our suggested way of adding a dependency is to run `poetry lock` with `--no-update` flag. That way we don't get unexpected version updates of subdependencies.

File: .github/workflows/main.yaml [hunk 4]
```diff
@@ -167,7 +167,7 @@ jobs:
       - name: Run base Odoo tests
         run: |
           export ODOO_IMAGE=${{ needs.combine-docker.outputs.IMAGE_REV }}
-          docker compose -f tests/docker-compose.yml run --rm --remove-orphans odoo ./odoo-bin -d test --test-enable --stop
+          bash tests/run_tests.sh
```
Comment (borna-ng) [inline .github/workflows/main.yaml]: It would be cool to have the file executable.

**Review by borna-ng — APPROVED** [review]
Works very well 🚀

I just have a couple of improvement suggestions

File: pyproject.toml (hunk 2 above)
Comment (jhelfferich) [reply pyproject.toml]: Great suggestion :+1: Done.

File: tests/run_tests.sh (hunk 3 above)
Comment (jhelfferich) [reply tests/run_tests.sh]: Done

File: poetry.lock
Comment (jhelfferich) [reply poetry.lock:1]: Done

File: .github/workflows/main.yaml (hunk 4 above)
Comment (jhelfferich) [reply .github/workflows/main.yaml]: Done

**Pull request merged**